### ISSUE TITLE
On-demand callback not called for menu items in Racket 7.6

### ISSUE THREAD

**Issue by alex-hhh** [racket/gui#168]
The program below illustrates the problem.  When I click on the "File" menu, I expect the `on-demand` callback to be invoked and "on-demand called" to be printed to the console.  The "on-demand" callback is not invoked in Racket 7.6, so nothing is printed to the console in this case.

The code works correctly in Racket 7.5.  I am running Windows 10.

```racket
#lang racket/gui

(define (on-demand m) (printf "on-demand called~%"))
(define (on-menu m e) (printf "selected menu item: ~a~%" (send m get-label)))

(define toplevel (new frame% [label "Test"] [width 300] [height 200]))
(define menu-bar (new menu-bar% [parent toplevel]))
(define file-menu (new menu%
                       [parent menu-bar]
                       [label "&File"]
                       [demand-callback on-demand]))
(new menu-item% [parent file-menu] [label "Item 1..."] [callback on-menu])
(new menu-item% [parent file-menu] [label "Item 2..."] [callback on-menu])
(new menu-item% [parent file-menu] [label "Item 3..."] [callback on-menu])
(send toplevel show #t)
```

**Comment by alex-hhh** [comment]
The `on-demand` callback is also not invoked on Racket 7.6 CS.

**Comment by alex-hhh** [comment]
The problem is introduced in this commit:

https://github.com/racket/gui/commit/823e940809e41db54f9d14abdef3288d4554388d

The problem is fixed if I comment out the form at 

https://github.com/racket/gui/blob/823e940809e41db54f9d14abdef3288d4554388d/gui-lib/mred/private/wx/win32/queue.rkt#L159

**Comment by mflatt** [comment]
Thanks for tracking down the problem! I've pushed a change that I hope will fix it. (It does on my machine, at least.)

**Comment by alex-hhh** [comment]
I verified this is fixed in version 7.7.0.1--2020-04-13(9c53358/a).  Thanks fix fixing it,
